### DISCUSSION TITLE
[keyball39]右手の縦一列が入力出来ない

### DISCUSSION THREAD

**Discussion by shojikun777** [Yowkees/keyball#458] [Q&A] [answered]
年末の最中に申し訳ございません。
電子工作初心者です。
他の方のQAを参考に下記対応をしてみましたが改善出来ず、ご支援いただきたくお願い申し上げます。
（発生現象）
・右手側の縦一列のキー４つが入力出来ません（写真のマスキング箇所）
（実施事項）
・キーソケットはんだ付け直し（上記以外の他の2か所はこれで治りました）
・ProMicro用12ピンソケットとPCBとのはんだ付け直し、ショート目視確認
・ProMicro用12PピンヘッダとProMicroとのはんだ付け直し、ショート目視確認
・ProMicroの左右スワップ（OLEDモジュールはスワップせず）

![PXL_20231227_051132965 MP](https://github.com/Yowkees/keyball/assets/154955752/d7b42f76-6fe4-43b0-b444-fcdc34df5569)
![PXL_20231227_052833626](https://github.com/Yowkees/keyball/assets/154955752/06846e1c-e826-4652-8165-c07d99c7684b)

**Comment by Ryo624** [comment]
ご連絡ありがとうございます。

キーの症状について了解しました。
はんだ付けの直し等も実施していただきましてありがとうございます。

今回の縦一列の反応が無いキーはCOL0という信号線とつながっていますので、写真に示しました赤点の端子の裏と表両方のはんだ付けをもう一度やり直していただいてもよろしいでしょうか。隣の端子とショートしていないかをよく確認していただけますと幸いです。
![292968450-06846e1c-e826-4652-8165-c07d99c7684b](https://github.com/Yowkees/keyball/assets/143698349/54703864-1d0a-4db3-8185-a84f69527b8a)

また、基板にはんだのカスが飛び散っているように見えますので、基板にあります小さな穴に入り込むと断線等の原因になりますので、お手数ですがこちらの除去も併せて実施していただけますでしょうか。

**Comment by shojikun777** [comment]
Ryo624さま
ご支援ありがとうございます。
・COL0の端子の裏と表両方のはんだ付けやり直し
・出来る範囲ではんだのカス除去（スマホカメラ5倍ズーム）
実施致しましたが現象変わらずでした...すみません。

お伝えしていなかったことで、LED実装しておりますが写真の通りかなり点灯しない状況です。
こちらは現時点で諦めており、キーボードとして利用出来ればよいと考えています。
こちらのLEDトラブルが上記に影響している可能性があるかも知れません。
![PXL_20231227_123303478](https://github.com/Yowkees/keyball/assets/154955752/979aff8f-9035-42da-a0b1-d304f0d80cbe)
![PXL_20231227_123252425](https://github.com/Yowkees/keyball/assets/154955752/66d8108c-93ea-471a-a97c-668c8fba97a2)
![PXL_20231227_122701966 MP](https://github.com/Yowkees/keyball/assets/154955752/31302a67-2796-489e-ac15-865cce1adb54)
![PXL_20231227_122544707](https://github.com/Yowkees/keyball/assets/154955752/4a7ea274-a30f-4df7-ae5a-cee52f565351)

**Comment by Ryo624** [comment]
はんだ付け等のやり直しを実施していただきましてありがとうございます。

写真に示しましたキーソケットの赤点とProMicroのCOL０の赤点が導通しているかを確認していただけますでしょうか。
![292968450-06846e1c-e826-4652-8165-c07d99c7684b](https://github.com/Yowkees/keyball/assets/143698349/e6e9e662-403b-4bac-9f12-cfd6a974ba42)

また、キーソケットのはんだ付けが多すぎるので、はんだ吸い取り線を使用してはんだの量を減らしてみてください。
該当するキーソケットの付近にはLEDにつながる線が通っていますので、LEDが点灯しない原因にもなっているかもしれません。

**Comment by shojikun777** [comment, marked as answer]
ご連絡が遅れ申し訳ありません
テスターが届いたので確認しました
①キーソケットの赤点とProMicroのCOL０の赤点は導通していました
②キーソケット両端子はキーを押下すると導通していました
③メモ帳を立上げキー入力を確認するも、復旧せず右端ABCDキー入力出来ず
④RESETスイッチを素早く2回押してみました
⑤Remapでキー設定を全てリセットし、事前にセーブしたデータをリロード
⑥メモ帳を立上げキー入力を確認するも、復旧せず右端ABCDキー入力出来ず
⑦キーソケットのはんだ吸い取りはフラックスが明日届くので実施予定です

PCB上の結線は良さそうな感じですが、何か対応すべきことはございますでしょうか。

![PXL_20231229_101510558](https://github.com/Yowkees/keyball/assets/154955752/a72fa0d2-3266-458e-b292-1322bf625733)

**Comment by shojikun777** [comment]
Ryo624さま
その後、フラックスが届いたので以下の対応をしたところ原因が？ですが「復旧」致しました！
①キーソケットのはんだ吸い取り、はんだやり直し
②LEDのはんだやり直し
③Remapでキーアサイン修正
　※今までキーアサインのドラッグアンドドロップが出来なかったのですが、何故か出来るようになりました。
④テキストでキー入力したところ、治っていました。

道具の準備不足（フラックス・フラックスクリーナ・はんだ吸い取り器）からつまずき、ご迷惑をおかけいたしました。
年末の最中、色々とご支援ありがとうございました。
これから楽しく使い込んでゆこうと思います。

**Reply by Ryo624** [reply]
shojikun777様

様々な対策を実施していただきまして、無事に解決していただきましてありがとうございます！
ぜひ末永くご愛用していただければ幸いです。
